### DISCUSSION TITLE
Why isn't it rendering?

### DISCUSSION THREAD

**Discussion by Rohcode** [jeffreylanters/react-unity-webgl#245] [Help Wanted!]
Can anyone please advise how to resolve what's going wrong here?

From Unity 2021, 'Build and Run' serves the files in a browser and works fine.  
I import Unity's Build files to a basic React app, set paths (/public/) and used the demo code. Yet the webgl component stays black and throws some errors. 
<img width="1400" alt="Screen Shot 2021-09-28 at 3 18 02 pm" src="https://user-images.githubusercontent.com/38774992/135027516-f4012ffe-4016-4990-b367-b0b861d036ca.png">
<img width="1643" alt="Screen Shot 2021-09-28 at 3 05 12 pm" src="https://user-images.githubusercontent.com/38774992/135027568-e1cb91bc-06a4-4078-8aaa-719b2203f115.png">

Prior error - WebGL-0x7fd8d3df0800]GL ERROR :GL_INVALID_VALUE : glTexStorage2D: dimensions out of range -

**Comment by jeffreylanters** [comment]
Hi. Can you validate your canvas is actually being displayed on your screen?

**Comment by Rohcode** [comment]
Hi Jeffrey
Yes sir. When I start the App the canvas is white (as seen in image 1). The console then logs that the loader.js has served .framework.js and the canvas turns black (2) as the console keeps logging various issues (3).
It seems to be running but not rendering correctly (just shows black instead of the correct rendering - which Unity serves on 'Build and Run' from a Unity project. Please let me know if you can recommend ways to approach troubleshooting such an issue. 
Thanks 
 
 
<img width="1636" alt="1" src="https://user-images.githubusercontent.com/38774992/135179888-4f1877df-258b-4785-a7a6-a208ab654cdd.png">
<img width="1634" alt="2" src="https://user-images.githubusercontent.com/38774992/135179898-3c82d78f-0cb5-42df-8ec5-e470a9b806d6.png">
<img width="1637" alt="3" src="https://user-images.githubusercontent.com/38774992/135179911-c3ff5dc5-3f6a-499e-87e1-f769da79d711.png">

**Comment by jeffreylanters** [comment]
Hi. Could you try providing a width and height for the canvas using styles or a className? Looking at your inspector, I think I know what's going wrong haha. The canvas should not scale itself to these enormous values. I will look into this, which browser and OS are you using?

![image](https://user-images.githubusercontent.com/3404617/135235500-81246fe4-f2dd-43d7-baa8-99f4ff5104b6.png)

Let me know if setting the dimensions solves your problem.

**Comment by Rohcode** [comment]
Ok thanks, I'm using Brave, Chrome, Safari, Firefox and OSX on a Macbook. 

I tried wrapping with width and height (not sure if there's a better way?), by returning:
`<canvas width="1000" height="330">
<Unity unityContext={unityContext} />
</canvas>`
Don't think this wrapped correctly (given the nested canvas in screenshot). It throws a size error (screenshot). 
<img width="1379" alt="Screen Shot 2021-09-30 at 10 12 58 am" src="https://user-images.githubusercontent.com/38774992/135365411-87be94d4-b67c-4b44-b20b-129aeeaaf6ee.png">
<img width="1381" alt="Screen Shot 2021-09-30 at 10 10 46 am" src="https://user-images.githubusercontent.com/38774992/135365338-9984c52d-e5d6-4929-a37c-215a658cf0f4.png">
Am trying to learn more now about styles and what the defaults are for how sizing and scaling. Please lmk if you have any ideas.

**Comment by jeffreylanters** [comment]
No, I meant adding these style properties to the Unity component itself. More like this:

(See the documentation for further details)

```jsx
import React from "react";
import Unity, { UnityContext } from "react-unity-webgl";

const unityContext = new UnityContext({
  loaderUrl: "build/myunityapp.loader.js",
  dataUrl: "build/myunityapp.data",
  frameworkUrl: "build/myunityapp.framework.js",
  codeUrl: "build/myunityapp.wasm",
});

function App() {
  return (
    <Unity
      unityContext={unityContext}
      style={{
        height: 600,
        width: 950,
      }}
    />
  );
}

```

**Reply by jeffreylanters** [reply]
@Rohcode Were you able to test this solution? I'd like to hear if it worked.

**Comment by Rohcode** [comment]
Yes, thank you @jeffreylanters. It worked in the demo project.
I've yet to get it working however in my TS production project, which throws the error:
`app.js:56454 A problem occurred while mounting Unable to load Build/1.loader.js [object Event]`
Thought it was a paths issue so reworked my demo  project code to match the paths of my prod, tested and the demo works but prod not. I do recall prior issues using using a public TS folder so learning more about this now. Please lmk if you have any suggestions because I'm keen to learn of any.
Thanks again
